### PR TITLE
feat: extract all website-cleaning into the new code-based website-cleaner tool

### PR DESCRIPTION
**Pull request by iwilltry42**
Note: this is using a plain http client for LLM requests - this will change as soon as we know where else to put the retry logic.

The new flow makes it such that
1. the website data-source only scrapes websites and saves them unmodified as html
2. the new website-cleaner tool takes care of cleaning the html programatically, convert it to markdown and (if small enough), have an LLM clean it again

Ref https://github.com/otto8-ai/otto8/issues/753

### CODE REVIEW

File: website-cleaner/main.go [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+	}
+
+	markdownSize := len(markdown)
+
+	if markdownSize < 100_000 {
+		logErr.Info("Markdown content is less than 100,000 characters -> sending to LLM for further cleanup")
+
+		prompt := "The following content is a scraped webpage converted to markdown. Please remove any content that came from the website header, footer, or navigation. The output should focus on just the main content body of the page. Maintain the markdown format, including any links or images.\n\n" + markdown
+
+		url := fmt.Sprintf("%s/chat/completions", os.Getenv("OPENAI_BASE_URL"))
```
[the remark this answers is not in the document]
Comment (iwilltry42) [reply website-cleaner/main.go:105]: llm-proxy in Otto8

File: website-cleaner/main.go [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+
+	markdown, err := converter.ConvertString(html)
+	if err != nil {
+		logOut.WithError(fmt.Errorf("failed to convert html to markdown: %v", err)).Error()
+		os.Exit(0)
+	}
+
+	markdownSize := len(markdown)
+
+	if markdownSize < 100_000 {
```
[the remark this answers is not in the document]
Comment (StrongMonkey) [reply website-cleaner/main.go:100]: how did we end up determine the threshold of 100000?

File: website-cleaner/main.go [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, 10*time.Minute)
+		defer cancel()
+
+		model := os.Getenv("OTTO8_DEFAULT_LLM_MINI_MODEL")
+		if model == "" {
+			logOut.WithError(fmt.Errorf("OTTO8_DEFAULT_LLM_MINI_MODEL is empty")).Error()
+			os.Exit(0)
+		}
```
Comment (thedadams) [inline website-cleaner/main.go:120]: Darren explicitly asked that we default to something reasonable in this case. I think `gpt-4o-mini` would be good here.

File: website-cleaner/main.go [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+	const baseDelay = time.Millisecond * 200
+	var resp *http.Response
+	var err error
+
+	var failures []string
+
+	// Save the original request body
+	var bodyBytes []byte
+	if req.Body != nil {
+		bodyBytes, err = io.ReadAll(req.Body)
```
Comment (thedadams) [inline website-cleaner/main.go:225]:
```suggestion
		bodyBytes, err = io.ReadAll(req.Body)
		_ = req.Body.Close()
```

File: website-cleaner/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (iwilltry42) [reply website-cleaner/main.go:100]: @ibuildthecloud set it at some point.. no idea :)

[the remark this answers is not in the document]
Comment (iwilltry42) [reply website-cleaner/main.go:100]: @g-linville moved the function around :+1:

**Review by g-linville — APPROVED** [review]

**Review by thedadams — APPROVED** [review]

**Pull request merged**